worker/lease/manager.go

```diff
@@ -150,21 +160,20 @@ func (manager *Manager) loop() error {
 
 	defer manager.wg.Wait()
 	blocks := make(blocks)
+	manager.setupInitialTimer()
 	for {
 		if err := manager.choose(blocks); err != nil {
+			manager.config.Logger.Tracef("[%s] exiting main loop with error: %v", manager.logContext, err)
 			return errors.Trace(err)
 		}
-
-		leases := manager.config.Store.Leases()
-		for leaseName := range blocks {
-			if _, found := leases[leaseName]; !found {
-				manager.config.Logger.Tracef("[%s] unblocking: %s", manager.logContext, leaseName)
-				blocks.unblock(leaseName)
-			}
-		}
 	}
 }
 
+func (manager *Manager) lookupLease(leaseKey lease.Key) (lease.Info, bool) {
+	lease, exists := manager.config.Store.Leases(leaseKey)[leaseKey]
```

Comment:
Just return it directly?

Reply:
You can't because Go is doing a "magic" 2 value return from a map, but that doesn't seem to work as a magic return directly from a function (it treats it as a one-value return, AFAICT).

./manager.go:173:2: not enough arguments to return
        have (lease.Info)
        want (lease.Info, bool)